aikau/src/main/resources/alfresco/services/actions/ChangeTypeService.js

```diff
@@ -0,0 +1,326 @@
+ * are expected to be provided as [a configuration attribute]{@link module:alfresco/services/actions/ChangeTypeService#types} 
+ * when instantiating this service. When used on Alfresco Share the types can be found in the XML configuration 
+ * files.
+ *
+ * @example <caption>This is an example configuration:</caption>
+ * {
+ *   name: "alfresco/services/actions/ChangeTypeService",
+ *   config: {
+ *     types: [
+ *       {
+ *         name: "cm:folder"
+ *       },
+ *       {
+ *         name: "cm:content",
+ *         label: "Content",
+ *         subTypes: [
+ *           {
+ *             name: "cm:sub-content",
+ *             label: "Sub-Content",
+ *               subTypes: [
+ *                 {
+ *                   name: "cm:super-sub-content"
+ *                 }
+ *               ]
+ *            }
+ *         ]
+ *       }
+ *     ]
+ *   }
+ * }
+ *
+ * @module alfresco/services/actions/ChangeTypeService
+ * @extends module:alfresco/services/BaseService
+ * @mixes module:alfresco/core/CoreXhr
+ * @author Dave Draper
+ * @since 1.0.58
+ */
+define(["dojo/_base/declare",
+        "alfresco/services/BaseService",
+        "alfresco/core/CoreXhr",
+        "alfresco/core/topics",
+        "service/constants/Default",
+        "dojo/_base/lang",
+        "dojo/_base/array",
+        "alfresco/forms/controls/Select"],
+        function(declare, BaseService, AlfCoreXhr, topics, AlfConstants, lang, array) {
+
+   return declare([BaseService, AlfCoreXhr], {
+
+      /**
+       * An array of the i18n files to use with this widget.
+       *
+       * @instance
+       * @type {object[]}
+       * @default [{i18nFile: "./i18n/ChangeTypeService.properties"}]
+       */
+      i18nRequirements: [{i18nFile: "./i18n/ChangeTypeService.properties"}],
+
+      /**
+       * This is the array of available types that are available. 
+       *
+       * @instance
+       * @type {array}
+       * @default
+       */
+      types: null,
+
+      /**
+       * Registers change type subscriptions.
+       *
+       * @instance
+       * @listens module:alfresco/core/topics#CHANGE_TYPE_REQUEST
+       */
+      registerSubscriptions: function alfresco_services_actions_ChangeTypeService__registerSubscriptions() {
+         this.alfSubscribe(topics.CHANGE_TYPE_REQUEST, lang.hitch(this, this.onChangeTypeRequest));
+      },
+
+      /**
+       * Handles requests to change the type of a node. This will perform an XHR request if necessary to load the full
+       * metadata for the node to be upated in order to retrieve its current type.
+       * 
+       * @instance
+       * @param {object} item The item to perform the action on
+       * @fires module:alfresco/core/topics#GET_DOCUMENT
+       */
+      onChangeTypeRequest: function alfresco_services_actions_ChangeTypeService__onChangeTypeRequest(payload) {
+         if (payload && payload.node && payload.node.type)
```

Comment:
These payload attributes (and others in this function) aren't defined in `topics.js` against `CHANGE_TYPE_REQUEST`. Is that intentional?